marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.mjs

```diff
@@ -408,12 +409,34 @@ function validateAndTestMapping(mapping, uri) {
   //modify the document instance with the properly updated one if a pre-step interceptor was used
   let newDocumentIsLoad = false;
   if (mapping.name) {
+    let docActual = cts.doc(uri);
     const updatedDocument = invokeGetDocument(mapping.name, uri);
+    sourceDocument = cts.doc(uri).toObject();
     if (updatedDocument && updatedDocument.format === "JSON") {
-      sourceDocument = cts.doc(uri).toObject();
       if (sourceDocument) {
         sourceDocument = updatedDocument.data;
         sourceDocument = xdmp.toJSON(sourceDocument);
+        newDocumentIsLoad = true;
+      }
+    }else{
+      if (updatedDocument && updatedDocument.format === "XML") {
+        const updatedData = updatedDocument.data;
+        let xmlTransform = json.transformFromJson(updatedData, json.config("custom"));
+        sourceDocument = docActual;
+        xdmp.eval(`
+          declareUpdate();
+          const sourceDocument = cts.doc('${uri}');
+          const xmlTransform = xdmp.unquote("${xmlTransform}");
+          const instanceNodes = sourceDocument.xpath('/envelope/instance');
+
+          if (instanceNodes.length > 0) {
+            const instanceNode = instanceNodes[0];
+            xdmp.nodeReplace(instanceNode, xmlTransform);
```

Comment:
We shouldn't be updating content in the database as part of testing/validating data.